layers-samples/proj-layer.html

```diff
@@ -0,0 +1,258 @@
+        xrRefSpace = refSpace;
+        proj_layer = xrGLFactory.createProjectionLayer({ space: refSpace, stencil: false });
+        session.updateRenderState({ layers: [proj_layer] });
+
+        session.requestAnimationFrame(onXRFrame);
+      });
+    }
+
+    function onEndSession() {
+      xrSession.end();
+    }
+
+    function onSessionEnded(event) {
+      if (event.session.isImmersive) {
+        xrButton.setSession(null);
+      }
+      xrSession = null;
+
+      // In this simple case discard the WebGL context too, since we're not
+      // rendering anything else to the screen with it.
+      gl = null;
+    }
+
+    function onXRFrame(time, frame) {
+      let pose = frame.getViewerPose(xrRefSpace);
+      xrSession.requestAnimationFrame(onXRFrame);
+
+      if (pose) {
+        gl.bindFramebuffer(gl.FRAMEBUFFER, xrFramebuffer);
+
+        // This is a different rendering pattern than the previous samples
+        // used, but it should be more efficent. It's very common for apps
+        // being ported to XR to take existing 2D rendering code and call the
+        // top-level "drawScene" function once per XR view, effectively
+        // drawing a single eye at a time. However, this causes many state
+        // changes to be duplicated, which adds to the render loop's overhead.
+        // By providing the matrices and viewports as an array to the drawing
+        // function it can do all the necessary binding once and then call the
+        // actual draw commands in a tighter loop, only changing the matrices
+        // and viewport each time. This does mean that the viewport is changed
+        // much more frequently (N times per object instead of N times per
+        // scene) but it's typically a pretty cheap thing to change and will
+        // almost always be easily outweighed by the savings from not
+        // redundantly binding everything else.
+
+        // For example, a traditional draw loop would do this:
+
+        // Draw(views):
+        //   for each view in views:
+        //     setViewport();
+        //     for each object in scene:
+        //       bindProgram();
+        //       bindMatrices();
+        //       bindUniforms();
+        //       bindBuffers();
+        //       bindTextures();
+        //       draw();
+
+        // While this method results in a loop more like this:
+
+        // Draw(views):
+        //   for each object in scene:
+        //     bindProgram();
+        //     bindUniforms();
+        //     bindBuffers();
+        //     bindTextures();
+        //     for each view in views:
+        //       setViewport();
+        //       bindMatrices();
+        //       draw();
+
+        // Note that for the complexity of the scene in this samples this
+        // won't make much visible performance difference, but we're using the
+        // more efficient pattern anyway as a way of promoting best practices.
```

Comment:
This entire comment block is unnecessary for this sample, as it's covered in a more focused manner in a separate sample. Let's remove it so that readers of this code can stay focused on the projection layer aspects.